### DISCUSSION TITLE
Repeat Comments Vim Scripts Not Working with Rider

### DISCUSSION THREAD

**Discussion by Marchonq** [JetBrains/ideavim#782] [Q&A] [answered]
I have been using Rider with IdeaVim for a while, but suddenly after an update, my repeat command vim scripts is not functioning. I tested the same script in PyCharm and it works fine. I don't know where is the problem.
I am using Rider 2023.3 and my Vim Plugin is 2.7.5. Here is my vim scripts and mapping
```
function! CommentMulti(total)
    let c = 0
    while c < a:total
      execute 'action CommentByLineComment'
      let c += 1
    endwhile
endfunction

" mappings:
nmap <leader>2c :call CommentMulti(2)<CR>
nmap <leader>3c :call CommentMulti(3)<CR>
```

**Comment by lippfi** [comment, marked as answer]
AFAIK, nothing has changed from the Vim Script perspective, and it seems to be an issue with action execution in the IDE. I've created a ticket to investigate this issue: https://youtrack.jetbrains.com/issue/VIM-3229/Repeat-Comments-Vim-Scripts-Not-Working-with-Rider.

IdeaVim supports the Commentary plugin, as detailed here: https://github.com/JetBrains/ideavim/wiki/IdeaVim-Plugins#commentary. Will this be sufficient for your needs?

P.S. There is no need to pass action CommentByLineComment to the execute command; action is already a Vim command.
P.P.S. The CommentByLineComment action works with selections. You could select total lines and then execute the action once.

**Reply by Marchonq** [reply]
Thanks for your kindly reply. I decided to use the Commtary plugin to work with comments instead.
